cmd/vcert/commands.go

```diff
@@ -318,73 +346,127 @@ func doCommandGetcred1(c *cli.Context) error {
 		return fmt.Errorf("could not create TPP connector: %s", err)
 	}
 
-	if cfg.Credentials.RefreshToken != "" {
-		resp, err := tppConnector.RefreshAccessToken(&endpoint.Authentication{
-			RefreshToken: cfg.Credentials.RefreshToken,
-			ClientId:     flags.clientId,
-			Scope:        flags.scope,
-		})
-		if err != nil {
-			return err
+	switch c.Command.Name {
+	case commandGetCredName:
+		//TODO: quick workaround to supress logs when output is in JSON.
+		if flags.credFormat != "json" {
+			logf("Getting credentials...")
 		}
-		if flags.credFormat == "json" {
-			jsonData, err := json.MarshalIndent(resp, "", "    ")
+
+		if cfg.Credentials.RefreshToken != "" {
+			resp, err := tppConnector.RefreshAccessToken(&endpoint.Authentication{
+				RefreshToken: cfg.Credentials.RefreshToken,
+				ClientId:     flags.clientId,
+				Scope:        flags.scope,
+			})
+			if err != nil {
+				return err
+			}
+			if flags.credFormat == "json" {
+				jsonData, err := json.MarshalIndent(resp, "", "    ")
+				if err != nil {
+					return err
+				}
+				fmt.Println(string(jsonData))
+			} else {
+				tm := time.Unix(int64(resp.Expires), 0).UTC().Format(time.RFC3339)
+				fmt.Println("access_token: ", resp.Access_token)
+				fmt.Println("access_token_expires: ", tm)
+				fmt.Println("refresh_token: ", resp.Refresh_token)
+			}
+		} else if cfg.Credentials.User != "" && cfg.Credentials.Password != "" {
+			resp, err := tppConnector.GetRefreshToken(&endpoint.Authentication{
+				User:     cfg.Credentials.User,
+				Password: cfg.Credentials.Password,
+				Scope:    flags.scope,
+				ClientId: flags.clientId})
 			if err != nil {
 				return err
 			}
-			fmt.Println(string(jsonData))
+			if flags.credFormat == "json" {
+				jsonData, err := json.MarshalIndent(resp, "", "    ")
+				if err != nil {
+					return err
+				}
+				fmt.Println(string(jsonData))
+			} else {
+				tm := time.Unix(int64(resp.Expires), 0).UTC().Format(time.RFC3339)
+				fmt.Println("access_token: ", resp.Access_token)
+				fmt.Println("access_token_expires: ", tm)
+				if resp.Refresh_token != "" {
+					fmt.Println("refresh_token: ", resp.Refresh_token)
+				}
+			}
```

Comment:
can this block of code be converted to a common func? I see that is used in more than one place.

Reply:
The `else` block varies by which action is specified (`getcred`, `checkcred`, and `voidcred`).  The code for the JSON output is common but it is only 5 lines on code (3 for checking error) so I didn't think it was necessary.  Do you think so?

Reply:
yes, I though the same, but also I though that if the code for the output changes/increment, then could be more easy to do the change only in one place reducing the risk to forgot to do the change in the places where the code logic is common. that is why I'm asking about if you consider to do that change :-)

Reply:
Added `outputJSON` function to address